src/Shared/Buffers.MemoryPool/DefaultMemoryPoolFactory.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Connections;
+
+namespace Microsoft.AspNetCore;
+
+#nullable enable
+
+internal sealed class DefaultMemoryPoolFactory : IMemoryPoolFactory<byte>, IAsyncDisposable
+{
+    private readonly IMeterFactory _meterFactory;
+    private readonly ConcurrentDictionary<PinnedBlockMemoryPool, PinnedBlockMemoryPool> _pools = new();
+    private readonly PeriodicTimer _timer;
+    private readonly Task _timerTask;
+
+    public DefaultMemoryPoolFactory(IMeterFactory? meterFactory = null)
+    {
+        _meterFactory = meterFactory ?? NoopMeterFactory.Instance;
```

Comment:
This looks like it would only be null in tests? Could it be made to always expect a meter factory, and then test code is responsible for creating the factory with a test meter factory?